### PR TITLE
Update Product.html

### PR DESCRIPTION
**Pull request by kul3r4**
This adds an additional note about default variables. It also updates the documentation.
@kmh287 I added this after #747

### CODE REVIEW

File: src/50_Samples_%26_Templates/Product.html [hunk 1]
```diff
@@ -387,7 +387,9 @@ <h2 class="m1">Apple</h2>
    <!-- ## Image Gallery -->
   <!-- The `amp-carousel` component works very well for product image galleries. Learn how the `amp-carousel` component works [here](/advanced/image_galleries_with_amp-carousel/).
 
-  Notice how we are binding to the `slide` and `class` by using to the values of the variables `selectedSlide` and `selectedColor`. -->
+  Notice how we are binding to the `slide` and `class` by using to the values of the variables `selectedSlide` and `selectedColor`. 
+  It's not possible to set a default value for variables, for example we use the expression`(!selectedColor || selectedColor == 'green')`
```
Comment (kmh287) [inline src/50_Samples_%26_Templates/Product.html]: This isn't quite right as `amp-state` elements can be used to define initial state. The difference is that all variables defined in an `amp-state` need to be prefixed by the `amp-state` element's id. It's not possible to specify initial values for variables at the top level.

Comment (kul3r4) [inline src/50_Samples_%26_Templates/Product.html]: Ok, so if `selectedColor` was part of amp-state and it had `green` as value, does it mean we would not need to check `<ul class="show" [class]="(!selectedColor || selectedColor == 'green') ? 'show' : 'hide'"`

Comment (kul3r4) [inline src/50_Samples_%26_Templates/Product.html]: I trieI tried to update the code this way:
`on="tap:AMP.setState({product.selectedSlide: 0})"`
and I get:

> cdn.ampproject.org/v0/amp-bind-0.1.js:22 amp-bind: AMP.setState() failed with error: Parse error on line 1:
> {product.selectedSlide: 0}
> --------^
> Expecting ':', got '.'​​​

Is it supported to set variables from the json state in the `setState`? I remember this was not happening at the start and this is why I was using global variables.

Comment (kmh287) [inline src/50_Samples_%26_Templates/Product.html]: Sorry, I misspoke earlier. In order to (re-) set the variable in a `setState` call you'll need to do.

`AMP.setState({product: {selectedSlide : 0}})`

Which will allow you to set `product.selectedSlide` to 0.

Comment (kul3r4) [inline src/50_Samples_%26_Templates/Product.html]: Ok,
I tried to rewrite it this way:
`          <amp-selector name="color"
                  layout="container"
                  [selected]="product.selectedColor"
                  on="select:AMP.setState({
                    {product: {
                      selectedColor: event.targetOption,
                      selectedSlide: 0,
                      selectedSize: product[event.targetOption].sizes[selectedSize] != 'unavailable' ? selectedSize : product[event.targetOption].defaultSize
                    }
                  })">`
but still I get:

> error.js:128 Invalid action definition in  
<amp-selector name=​"color" layout=​"container" [selected]=​"product.selectedColor" on=​"select:​AMP.setState({
                    {product:​ {
                      selectedColor:​ event.targetOption,
                      selectedSlide:​ 0,
                      selectedSize:​ product[event.targetOption]​.sizes[selectedSize]​ != 'unavailable' ? selectedSize :​ product[event.targetOption]​.defaultSize
                    }​
                  }​)​" class=​"i-amphtml-element i-amphtml-layout-container i-amphtml-layout i-amphtml-error" role=​"listbox" selected=​"green">​…​</amp-selector>​
 : [ select:AMP.setState({
                    {product: {
                      selectedColor: event.targetOption,
                      selectedSlide: 0,
                      selectedSize: product[event.targetOption].sizes[selectedSize] != 'unavailable' ? selectedSize : product[event.targetOption].defaultSize
                    }
                  }) ] ; unexpected token []

Comment (kmh287) [inline src/50_Samples_%26_Templates/Product.html]: Looks like there's an extra curly bracket at the beginning of the argument to `setState`

So it should be:
```js
on=​"select:​AMP.setState({
  product:​ {
    selectedColor:​ event.targetOption,
    selectedSlide:​ 0,
    selectedSize:​ product[event.targetOption]​.sizes[selectedSize]​ != 'unavailable' ? selectedSize :​ 
    product[event.targetOption]​.defaultSize
  }​
}​
```

**Comment by kul3r4** [comment]
sorry i closed this in error

**Pull request closed without merging**